buildifier/factory.bzl

```diff
@@ -137,16 +133,21 @@ def buildifier_impl_factory(ctx, *, test_rule):
             fail("Cannot use 'no_sandbox' without a 'workspace'")
         workspace = ctx.file.workspace.path
 
-    buildifier = ctx.toolchains["@buildifier_prebuilt//buildifier:toolchain"]._tool
-    out_file = ctx.actions.declare_file(ctx.label.name + ".bash")
+    toolchain = ctx.toolchains["@buildifier_prebuilt//buildifier:toolchain"]
+    buildifier = toolchain._tool
+    runner = toolchain._runner
+
+    out_ext = ".bash" if runner.label.name.endswith(".bash.template") else ".bat"
+    out_file = ctx.actions.declare_file(ctx.label.name + out_ext)
+
     substitutions = {
         "@@ARGS@@": shell.array_literal(args),
         "@@BUILDIFIER_SHORT_PATH@@": shell.quote(buildifier.short_path),
         "@@EXCLUDE_PATTERNS@@": exclude_patterns_str,
         "@@WORKSPACE@@": workspace,
     }
     ctx.actions.expand_template(
-        template = ctx.file._runner,
+        template = runner.files.to_list()[0],
```

Comment:
I'm not sure if this is the most idiomatic way... Feels a bit cumbersome.

Reply:
if this works it's fine with me but does `runner.file` work instead? 🤔

Reply:
Unfortunately no :( `Error: 'Target' value has no field or method 'file' (did you mean 'files'?)`